src/app/shared/input/input.component.ts

```diff
@@ -33,6 +33,8 @@ export class InputComponent implements OnInit {
 
   @HostListener('change', ['$event'])
   @HostListener('focusout', ['$event'])
+  @HostListener('click', ['$event'])
```

Comment:
jak bedziesz chcial kiedys ustawic swoj input na tryb obserwowania zmian "on blur" to bedziesz musial to dosc mocno zrefaktorowac. teraz angular musi nasluchiwac całej sterty eventow tylko po to zeby dowiedziec sie jaki jest stan kontrolki. a przeciez kontrolka ma observabla "statusChange" albo chociazby "valueChange" ktory bedzie sprawdzal ci te wlasciwosci kontrolki tylko przy zmianach statusu lub wartosci :) a nie przy kazdym evencie. A dodatkowo co jest tez bardzo zle - uzywasz metody "showErrors()" w szablonie html - co technicznie oznacza ze bedzie to wywolywane nawet wtedy kiedy poleci jakikolwiek inny event z innego miejsca